flang/include/flang/Optimizer/Dialect/FIRTypes.td

```diff
@@ -21,6 +21,32 @@ class FIR_Type<string name, string typeMnemonic> : TypeDef<fir_Dialect, name> {
   let mnemonic = typeMnemonic;
 }
 
+def ComplexType : FIR_Type<"Complex", "complex"> {
+  let summary = "Complex type";
+
+  let description = [{
+    Model of a Fortran COMPLEX intrinsic type, including the KIND type
+    parameter. COMPLEX is a floating point type with a real and imaginary
+    member.
+  }];
+
+  let parameters = (ins "KindTy":$kind);
+
+  let printer = [{
+    $_printer << "complex<" << getFKind() << '>';
+  }];
+
+  let genAccessors = 1;
+
+  let extraClassDeclaration = [{
+    using KindTy = unsigned;
+
+    KindTy getFKind() const { return getKind(); }
```

Comment:
How is `getKind()` auto-generated ? Are getters for parameters `xxx` auto-generated as `getXxx` ?

Reply:
Exactly. If you enable it with `let genAccessors = 1;` you have them. We can also change the internal variable name to `fKind` to have a single auto-generated accessor that matches the original function name.

Reply:
Background: The FKind (Fortran KIND) is here because at one point MLIR had a different definition of the term "kind" as related to the MLIR type system. (That may have disappeared.)

Reply:
Ok. Let's keep `FKind` for now and we can change it later if it's possible.

Reply:
Thanks for explaining, I was surprised because the ops accessor do not have the `get` prefix.